flocker/node/agents/ebs.py

```diff
@@ -1245,6 +1254,11 @@ def detach_volume(self, blockdevice_id):
         if ebs_volume.state != VolumeStates.IN_USE.value:
             raise UnattachedVolume(blockdevice_id)
 
+        if ebs_volume.attachments:
+            attachment_state = ebs_volume.attachments[0]['State']
+            if attachment_state == VOLUME_ATTACHMENT_BUSY:
+                raise DetachFailed(ebs_volume)
```

Comment:
This overall approach is "look before you leap", which is generally speaking less good than trying-and-failing. In this particular case, it's possible that although the volume _was_ busy on line 1258, it's no longer busy by line 1262, or that it's busy, but the symptoms are different. A more reliable way would be to catch whatever error is raised `_detach_ebs_volume`.